### PR TITLE
Search algorithms

### LINKED ISSUE #1722: [Entry] Artificial Intelligence: Search Algorithm
### Reason/inspiration (optional)

We seek an entry explaining the concept of a Search Algorithm in the context of Artificial Intelligence. 

### Entry Type (select all that apply)

New Topic Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by rykunk21**
### Description

Added an entry for search algorithms, search algorithms are an important concept when learning about artificial intelligence. closes #1722

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (caupolicandiaz) [inline content/general/concepts/search-algorithms/search-algorithms.md:24]:
```suggestion
Uninformed search algorithms consist of general purpose traversals of the state space or search tree without any information about how good a state is. These are also referred to as blind search algorithms.
```

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (caupolicandiaz) [inline content/general/concepts/search-algorithms/search-algorithms.md:28]:
```suggestion
Informed search algorithms have information about the goal during the traversal. This allows the search to prioritize its expansion toward the goal state instead of exploring directions that may yield a favorable cost but don't lead to the goal, or global optimum. By including extra rules that aid in estimating the location of the goal, known as _heuristics_, informed search algorithms can be more computationally efficient when searching for a path to the goal state.
```

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (caupolicandiaz) [inline content/general/concepts/search-algorithms/search-algorithms.md:54]:
```suggestion
### Pathfinding
```

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (caupolicandiaz) [inline content/general/concepts/search-algorithms/search-algorithms.md:58]:
```suggestion
### Optimization
```

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (caupolicandiaz) [inline content/general/concepts/search-algorithms/search-algorithms.md:62]:
```suggestion
### Game Playing
```

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (caupolicandiaz) [inline content/general/concepts/search-algorithms/search-algorithms.md:64]:
```suggestion
In game playing, search algorithms are used to evaluate all possible moves and choose the one that is most likely to lead to a win, or the best possible outcome. This is done by constructing a search tree where each node represents a game state and the edges represent the moves that can be taken to reach the associated new game state.
```

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (caupolicandiaz) [inline content/general/concepts/search-algorithms/search-algorithms.md:60]:
```suggestion
In optimization problems, the goal is to find the minimum or maximum value of a function, subject to some constraints. Search algorithms such as hill climbing or simulated annealing are often used in optimization cases.
```

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (caupolicandiaz) [inline content/general/concepts/search-algorithms/search-algorithms.md:56]:
```suggestion
Pathfinding problems involve finding the shortest path between two points in a given graph or network. BFS or A\* search can be used to explore a graph and find the optimal path.
```

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (caupolicandiaz) [inline content/general/concepts/search-algorithms/search-algorithms.md:52]:
```suggestion
Search algorithms are used in various fields of artificial intelligence, including:
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@rykunk21 I've finished my initial review. Please tag me when you've finished making updates. Thanks

**Comment by rykunk21** [comment]
@caupolicandiaz Thank you for the review, the updates should have been made at this point.

**Review by caupolicandiaz — APPROVED** [review]
@rykunk21 Looks good. This is ready for another review.

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (HishamT) [inline content/general/concepts/search-algorithms/search-algorithms.md:16]:
```suggestion
A **search algorithm** is a type of algorithm used in artificial intelligence to find the best or most optimal solution to a problem by exploring a set of possible solutions, also called a *search space*. A search algorithm filters through a large number of possibilities to find a solution that works best for a given set of constraints.
```

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (HishamT) [inline content/general/concepts/search-algorithms/search-algorithms.md:22]: This section and the one after it would look better if they were presented as bullet-points instead of section titles.

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (HishamT) [inline content/general/concepts/search-algorithms/search-algorithms.md:26]: See above comment.

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (HishamT) [inline content/general/concepts/search-algorithms/search-algorithms.md:40]:
```suggestion
This algorithm explores all the neighbor nodes at the current level before moving on to the nodes at the next level. Think of BFS as a traversal pattern that tries to explore broadly across many different paths at the same time.
```

**Review by HishamT — COMMENTED** [review]
Hey @rykunk21, I have finished the second review and left some comments.

**Comment by rykunk21** [comment]
@HishamT I think these updates are great, if you have anything else let me know.

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (HishamT) [inline content/general/concepts/search-algorithms/search-algorithms.md:22]:
```suggestion
- Uninformed Search: algorithms that are general purpose traversals of the state space or search tree without any information about how good a state is. These are also referred to as blind search algorithms.
```

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (HishamT) [inline content/general/concepts/search-algorithms/search-algorithms.md:24]: Remove this line.

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (HishamT) [inline content/general/concepts/search-algorithms/search-algorithms.md:26]:
```suggestion
Informed Search: algorithms that have information about the goal during the traversal, allowing the search to prioritize its expansion toward the goal state instead of exploring directions that may yield a favorable cost but don't lead to the goal, or global optimum. By including extra rules that aid in estimating the location of the goal (known as _heuristics_) informed search algorithms can be more computationally efficient when searching for a path to the goal state.
```
```

File: content/general/concepts/search-algorithms/search-algorithms.md
Comment (HishamT) [inline content/general/concepts/search-algorithms/search-algorithms.md:28]: Remove this line.

**Review by HishamT — COMMENTED** [review]
Looking pretty good. I did leave some more comments.

**Review by HishamT — APPROVED** [review]
Made a small change, but looks good!

**Comment by yangc95** [comment]
Excellent entry 👏 @rykunk21 live link: https://www.codecademy.com/resources/docs/general/search-algorithms

Screenshot 2023-03-19 at 5 06 54 PM

> It looks as if you're anonymous. If you want to be credited follow the instructions below to add your GitHub username to your Codecademy profile. If not, there is nothing more you need to do.
> 
> To add your GitHub username to your Codecademy profile:
> 
> 1.) Add your GitHub Username to your Codecademy Profile Details, you can get there on your desktop by pressing on your profile picture in the upper right-hand corner and selecting "Profile" in the drop-down box:
> 
> image
> 
> 2.) Then log out and log back into Codecademy using your GitHub login:
> 
> githublogin
>
> If the steps above do not work, it may be due to an email or local configuration issue.
>
> In addition, we may reach out in the future if any adjustments to these entries are required.

cc: @caupolicandiaz @HishamT

**Pull request merged**
